harness/determined/pytorch/_pytorch_trial.py

```diff
@@ -544,7 +541,8 @@ def _compute_validation_metrics(self) -> workload.Metrics:
             batch_metrics = []
 
             self.validation_loader = cast(torch.utils.data.DataLoader, self.validation_loader)
```

Comment:
while we're adjusting checks, can you convert this to an assertion rather than a cast?

(there's basically no cases where I think a cast is better than an assertion)

Reply:
You mean like `assert isinstance(self.validation_loader, DataLoader), "blah"`?

Reply:
yeah, but if it's a type check only for mypy's sake (because we can already guarantee that by this point in execution the validation_loader is set) then it doesn't need any message.

Reply:
Done.